juju/client/proxy/factory.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from juju.client.proxy.kubernetes.proxy import KubernetesProxy
+
+class ProxyTypeNotDefinedError(Exception):
+    pass
+
+class UnknownProxyTypeError(Exception):
+    pass
+
+class Factory(ABC):
+
+    @abstractmethod
+    def proxy_from_config(self, conf):
+        raise NotImplementedError()
+
+class JujuDataFactory(Factory):
+
+    def proxy_from_config(self, conf):
+        if conf is None:
+            return None
+
+        if "type" not in conf:
+            return None
+
+        proxy_type = conf['type']
+        if proxy_type != "kubernetes-port-forward":
+            raise UnknownProxyTypeError()
+
+        return self.__construct_kube_proxy(conf['config'])
+
+    def __construct_kube_proxy(self, config):
```

Comment:
FYI, I know Python supports this, but it's not common to use name-mangled, double-underscore prefix methods for privacy. You typically just use single-under-prefixed ones. This project uses single-underscore prefixed names for private methods (`rg 'def __[a-z]+\('` returns no results).